aten/src/ATen/cpu/vec256/vec256_base.h

```diff
@@ -390,24 +407,30 @@ inline T minimum(const T& a, const T& b) {
 // To save BC, it will not propagate NaN based on IEEE 754 201X
 template <class T> Vec256<T> inline clamp(const Vec256<T> &a, const Vec256<T> &min_vec, const Vec256<T> &max_vec) {
   Vec256<T> c = Vec256<T>();
+  using value_t = typename at::native::ztype<T>::value_t;
+  value_t (*zabs_)(T) = at::native::zabs;
   for (int i = 0; i != Vec256<T>::size(); i++) {
-    c[i] = a[i] < min_vec[i] ? min_vec[i] : (a[i] > max_vec[i] ? max_vec[i] : a[i]);
+    c[i] = zabs_(a[i]) < zabs_(min_vec[i]) ? min_vec[i] : (zabs_(a[i]) > zabs_(max_vec[i]) ? max_vec[i] : a[i]);
```

Comment:
Note for reviewers: `zabs` is no op on real only numbers. (The semantics of `zabs` is pretty weird. In particular, `zabs(-1) == -1` if `-1` is a non-complex type, but `== 1` if it is a complex type. That seems very fishy, semantically to me. I'd prefer not to implement clamp on complex unless we have a really good reason for hacking it up this way.

Reply:
Hmm, I was under the impression that the min, max arguments could be tensors, but the torch docs indicate that they are just numeric constants.

The following is a common voltage control mechanism that is commonly used in my work:
`np.where(np.where(np.abs(a) > np.abs(min), a, min) < np.abs(max), max)`

Perhaps I should just use `torch.where()`.